horovod/_keras/__init__.py

```diff
@@ -28,7 +28,8 @@
 def create_distributed_optimizer(keras, optimizer, name, device_dense, device_sparse,
                                  compression, sparse_as_dense, gradient_predivide_factor,
                                  op, backward_passes_per_step=1,
-                                 average_aggregated_gradients=False):
+                                 average_aggregated_gradients=False,
+                                 num_groups=0):
```

Comment:
I'm wondering if instead of `num_groups` we might want to name it `groups` in case we want to overload the semantics in the future.  For instance, allowing the user to provide explicits groups, or some other grouping criteria.  What do you think?  We can always cross the bridge if we come to it.

Reply:
I'd prefer to leave it `num_groups` for now for simplicity. I think we may want to use a target message size as an alternative grouping criteria in the future, but the argument name `groups` doesn't make a lot of sense in that context either, but rather `group_size_mb` or something like that. As far as explicitly defining groups, the list of gradient tensors within the optimizer are not easily accessible to the user at the time the `DistributedOptimizer` is constructed to form groups. However, more explicit control can still be achieved if desired through manual use of Horovod primitives instead of the `DistributedOptimizer` wrapper.

Reply:
Sounds good, that sounds reasonable to me.